src/modules/sensors/vehicle_gps_position/VehicleGPSPosition.cpp

```diff
@@ -129,7 +129,19 @@ void VehicleGPSPosition::Run()
 		_gps_blending.update(hrt_absolute_time());
 
 		if (_gps_blending.isNewOutputDataAvailable()) {
-			Publish(_gps_blending.getOutputGpsData(), _gps_blending.getSelectedGps());
+			sensor_gps_s gps_data = _gps_blending.getOutputGpsData();
+
+			if (!PX4_ISFINITE(gps_data.heading)) {
+				sensor_gps_s blended_gps_data = _gps_blending.getOutputBlendedGpsData();
```

Comment:
This still can result in `heading` being 0 since the `blended_gps_data.heading` field is populated only if a heading is availabing **and** and the blend weight is higher
https://github.com/PX4/PX4-Autopilot/blob/master/src/modules/sensors/vehicle_gps_position/gps_blending.cpp#L481-L489

Reply:
oh nvm I'm wrong